cpp/src/dual_simplex/cuts.cpp

```
@@ -0,0 +1,2860 @@
+{
+  FILE* fid = NULL;
+  fid       = fopen("solution.dat", "w");
+  if (fid != NULL) {
+    printf("Writing solution.dat\n");
+
+    std::vector<f_t> residual = lp.rhs;
+    matrix_vector_multiply(lp.A, 1.0, solution, -1.0, residual);
+    printf("|| A*x - b ||_inf %e\n", vector_norm_inf<i_t, f_t>(residual));
+    auto hash_combine_f = [](size_t seed, f_t x) {
+      seed ^= std::hash<f_t>{}(x) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+      return seed;
```

Cppcheck (2.19.0)

[error] Resource leak

(resourceLeak)